src/training/data.py

```diff
@@ -448,6 +483,36 @@ def get_csv_dataset(args, preprocess_fn, is_train, epoch=0, tokenizer=None, coll
     return DataInfo(dataloader, sampler)
 
 
+def get_hf_text_dataset(args, epoch=0, tokenizer=None, collate_fn=None):
+    dataset_name = args.flava_unimodal_mlm
+    assert dataset_name
+
+    dataset = HFTextDataset(
+        dataset_name,
+        split='train',
+        text_key='text',
+        tokenizer=tokenizer
+    )
+    num_samples = len(dataset)
+    sampler = DistributedSampler(dataset) if args.distributed else None
+    shuffle = sampler is None
+
+    dataloader = DataLoader(
+        dataset,
+        batch_size=args.flava_unimodal_mlm_batch_size,
+        shuffle=shuffle,
+        num_workers=args.workers,
+        pin_memory=True,
+        sampler=sampler,
+        collate_fn=collate_fn,
+        drop_last=True,
+    )
+    dataloader.num_samples = num_samples
+    dataloader.num_batches = len(dataloader)
```

Comment:
I don't like that we are adding attributes on the fly to the dataloader.

Reply:
I don't like it either, but it seems to be a thing that the OpenCLIP repo does a good amount (and the trainer expects it).